FeedAPIChallenge/Feed API/FeedItemsMapper.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  FeedItemsMapper.swift
+//  FeedAPIChallenge
+//
+//  Created by Mher Movsisyan on 08.08.21.
+//  Copyright © 2021 Essential Developer Ltd. All rights reserved.
+//
+
+import Foundation
+
+struct FeedItemsMapper {
+	struct Root: Decodable {
+		private let items: [Item]
+
+		var feedItems: [FeedImage] {
+			items.compactMap { item in
+				guard let uuid = UUID(uuidString: item.imageId),
+							let url = URL(string: item.imageURL) else { return nil }
+
+				return FeedImage(id: uuid, description: item.imageDesc, location: item.imageLoc, url: url)
+			}
+		}
+	}
+
+	struct Item: Decodable {
+		let imageId: String
+		let imageURL: String
+		var imageDesc: String?
+		var imageLoc: String?
+
+		enum CodingKeys: String, CodingKey {
+			case imageId = "image_id"
+			case imageURL = "image_url"
+			case imageDesc = "image_desc"
+			case imageLoc = "image_loc"
+		}
+	}
+
+	static let OK_200 = 200
+
+	static func map(data: Data, urlResponse: HTTPURLResponse) -> FeedLoader.Result {
+		if urlResponse.statusCode != FeedItemsMapper.OK_200 {
+			return .failure(RemoteFeedLoader.Error.invalidData)
+		}
+
+		guard let root = try? JSONDecoder().decode(Root.self, from: data) else {
+			return .failure(RemoteFeedLoader.Error.invalidData)
+		}
```

Comment:
It's possible to combine both the `if` and `guard` statements, since both paths return the same `invalidData` error:
```suggestion
		guard urlResponse.statusCode == OK_200,
		      let root = try? JSONDecoder().decode(Root.self, from: data) else {
			return .failure(RemoteFeedLoader.Error.invalidData)
		}
```
Also, `FeedItemsMapper.OK_200` can be simplified to `OK_200`, since this portion of code is already in a static context.